hack/local-up-cluster.sh

```diff
@@ -150,6 +151,11 @@ if [ "$(id -u)" != "0" ]; then
     echo "WARNING : This script MAY be run as root for docker socket / iptables functionality; if failures occur, retry as root." 2>&1
 fi
 
+if [ -z "$CLOUD_PROVIDER" ]; then
+    ENABLE_HOSTPATH_PROVISIONER=true
```

Comment:
The problem with this logic is there is no way to force disable the hostpath provisioner.

We should not set the default above, and then in this logic, check if it is unset as a condition. Then afterwards, set the false default if unset.